csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -431,17 +436,17 @@ class PredicateChcker : public IterVisitor {
           id->getParallelType() == ParallelType::Unswitch) {
         return true;
       }
+    }
 
-      // TODO: （Enable in a follow up)
-      //  This cannot yet be removed since smem initialization needs to be
-      //  handled specially, e.g. as in smem_reduce test. Will be able to
-      //  lift this one once the generic pred removal pass with fusion
-      //  traversal is ready.
-      auto consumer_def = consumer->definition();
-      if (ir_utils::isReductionOp(consumer_def)) {
-        if (producer->getMemoryType() == MemoryType::Shared) {
-          return true;
-        }
+    // TODO: (Enable in a follow up)
+    //  This cannot yet be removed since smem initialization needs to be
+    //  handled specially, e.g. as in smem_reduce test. Will be able to
+    //  lift this one once the generic pred removal pass with fusion
+    //  traversal is ready.
+    auto consumer_def = consumer->definition();
+    if (ir_utils::isReductionOp(consumer_def)) {
+      if (producer->getMemoryType() == MemoryType::Shared) {
+        return true;
```

Comment:
Unrelated to this PR, but I noticedt his check does not need to be inside the loop.